artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederationStream.java

```diff
@@ -106,4 +107,25 @@ public int getPriorityAdjustment() {
       return config.getConnectionConfiguration().getPriorityAdjustment();
    }
 
+   protected void callFederationStreamStartedPlugins() {
+      if (server.hasBrokerFederationPlugins()) {
+         try {
+            server.callBrokerFederationPlugins(plugin -> plugin.federationStreamStarted(this));
```

Comment:
If we are going to pass this to the plugin, we should extract interface, and method of federationStreamStarted should take the interface, not the implementation. As plugin should just have api's, else will cause issues in any future refactor (like you had done in last pr)